packages/ibm-products/src/components/InlineTip/InlineTip.test.js

```diff
@@ -0,0 +1,192 @@
+/**
+ * Copyright IBM Corp. 2023, 2023
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import { render, screen } from '@testing-library/react'; // https://testing-library.com/docs/react-testing-library/intro
+import userEvent from '@testing-library/user-event';
+
+import { pkg } from '../../settings';
+import uuidv4 from '../../global/js/utils/uuidv4';
+
+import { InlineTip, InlineTipButton, InlineTipLink } from '.';
+import InlineTipImage from './assets/inline-tip-image.png';
+
+const blockClass = `${pkg.prefix}--inline-tip`;
+const componentName = InlineTip.displayName;
+
+// values to use
+const title = `title (${uuidv4()})`;
+const children = `hello, world (${uuidv4()})`;
+const buttonLabel = `Click me (${uuidv4()})`;
+const tertiaryButtonLabel = `Show me (${uuidv4()})`;
+const linkLabel = `Learn more (${uuidv4()})`;
+const readMoreLabel = `Read more (${uuidv4()})`;
+const readLessLabel = `Read less (${uuidv4()})`;
+const className = `class-${uuidv4()}`;
+const dataTestId = uuidv4();
+
+describe(componentName, () => {
+  it('renders a component InlineTip', () => {
+    render(<InlineTip title={title}>{children}</InlineTip>);
+    expect(screen.getByRole('main')).toHaveClass(blockClass);
+  });
+
+  it.skip('has no accessibility violations', async () => {
+    const { container } = render(
+      <InlineTip title={title}>{children}</InlineTip>
+    );
+    await expect(container).toBeAccessible(componentName);
+    await expect(container).toHaveNoAxeViolations();
+  });
+
+  it(`renders children`, () => {
+    render(<InlineTip title={title}>{children}</InlineTip>);
+    screen.getByText(children);
+  });
+
+  it(`renders ghost button`, () => {
+    render(
+      <InlineTip
+        title={title}
+        contentCTA={
+          <InlineTipButton onClick={() => {}}>{buttonLabel}</InlineTipButton>
+        }
+      >
+        {children}
+      </InlineTip>
+    );
+    screen.getByText(buttonLabel);
+  });
+
+  it(`renders link`, () => {
+    render(
+      <InlineTip
+        title={title}
+        contentCTA={
+          <InlineTipLink href="https://www.ibm.com" target="_blank">
+            {linkLabel}
+          </InlineTipLink>
+        }
+      >
+        {children}
+      </InlineTip>
+    );
+    screen.getByText(linkLabel);
+  });
+
+  it(`renders icon button with 'x' icon`, () => {
+    render(
+      <InlineTip title={title} onClose={() => {}}>
+        {children}
+      </InlineTip>
+    );
+    expect(
+      document.querySelector('.c4p--inline-tip__close-icon')
```

Comment:
The `testing-library` way[^1] should be ...

```suggestion
      screen.getByRole('button', { name: 'Close' })
```

[^1]: https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-container-to-query-for-elements

Reply:
Thanks, testing is not one of my strong points :)

Reply:
Haha, also learning. I basically pull up the Kent C Dodds blog whenever I’m working on tests. 😂